### PR TITLE
Thunderscope: play/pause simulation button and shortcut

### PR DESCRIPTION
**Pull request by sarphion**
### Description
 A new toolbar added, which has a pause/play button. if pressed, the simulation will pause, and if it is in the pause state, and the button is pressed, the simulation resumes back to playing. 
    when button pressed, or the shortcut is used (ctrl+space), the button changes icon as well.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/thunderscope/thunderscope.py [hunk 1]
```diff
@@ -148,6 +148,21 @@ def __init__(
         self.window.setCentralWidget(self.tabs)
         self.window.setWindowTitle("Thunderscope")
 
+        self.is_playing = True
+
+        simulationToolBar = self.window.addToolBar("simulation")
+
+        pixmapi_pause = QStyle.StandardPixmap.SP_MediaStop
```
Comment (williamckha) [inline src/software/thunderscope/thunderscope.py]: Would a pause icon be better suited here? `SP_MediaPause`
A stop icon might be confused for halting/exiting the simulation

**Review by williamckha — COMMENTED** [review]
This is cool! Nice to see these features being added to thunderscope, will be helpful for a project I'm working on

**Review by itsarune — APPROVED** [review]
[ignore]

**Review by itsarune — CHANGES_REQUESTED** [review]
nice work getting it to work!

I'd like the toolbar to be right under the field widget: 
image

right now, you have to hunt down the toolbar. Also, I'd like the toolbar to be centred on its panel as well

File: src/software/thunderscope/thunderscope.py (hunk 1 above)
Comment (sarphion) [reply src/software/thunderscope/thunderscope.py]: I agree! Thanks! I changed it to a pause icon in my most recent commit

**Review by williamckha — DISMISSED** [review]

File: src/software/thunderscope/field/world_layer.py [hunk 2]
```diff
@@ -10,11 +10,11 @@
 from proto.geometry_pb2 import Point, Segment
 from software.py_constants import *
 from software.thunderscope.constants import (
-    BALL_HEIGHT_EFFECT_MULTIPLIER,
     LINE_WIDTH,
     SPEED_LINE_WIDTH,
     SPEED_SEGMENT_SCALE,
 )
+from software.thunderscope.constants import LINE_WIDTH
```
Comment (nimazareian) [inline src/software/thunderscope/field/world_layer.py:17]: Some of these changes seem to be made by accident when fixing merge conflicts and are unrelated to this ticket. They should be reverted to how they are on master.

File: src/software/thunderscope/field/world_layer.py [hunk 3]
```diff
@@ -403,13 +390,10 @@ def draw_ball_state(self, painter, ball_state: BallState):
...
         painter.setBrush(pg.mkBrush(Colors.BALL_COLOR))
 
-        # Ball should get larger as the height of the ball increases
-        ball_radius = BALL_MAX_RADIUS_METERS * (
-            1 + BALL_HEIGHT_EFFECT_MULTIPLIER * ball_state.distance_from_ground
+        painter.drawEllipse(
+            self.createCircle(ball_state.global_position, BALL_MAX_RADIUS_METERS)
         )
 
-        painter.drawEllipse(self.createCircle(ball_state.global_position, ball_radius))
```
Comment (nimazareian) [inline src/software/thunderscope/field/world_layer.py:411]: Likewise here with unwanted changes

File: src/software/thunderscope/thunderscope.py [hunk 4]
```diff
@@ -1,13 +1,22 @@
...
+# Unfortunately, PyQt6 doesn't install on Ubuntu 18. Thankfully both
+# libraries are interchangeable, and  we just need to swap them in this
+# one spot, and pyqtgraph will pick up on it and store the library under
+# pyqtgraph.Qt. So from PyQt5 import x becomes from pyqtgraph.Qt import x
+if "18.04" in platform.version():
+    import PyQt5
+    from PyQt5.QtWebEngineWidgets import QWebEngineView
+else:
+    import PyQt6
+    from PyQt6.QtWebEngineWidgets import QWebEngineView
```
Comment (nimazareian) [inline src/software/thunderscope/thunderscope.py:19]: Much of the changes in this file as well

File: src/software/thunderscope/thunderscope_main.py [hunk 5]
```diff
@@ -364,6 +364,11 @@ def __async_sim_ticker(tick_rate_ms):
                 SimulationState, simulation_state_buffer
             )
 
+            simulation_state_buffer = ThreadSafeBuffer(1, SimulationState)
+            tscope.simulator_proto_unix_io.register_observer(
+                SimulationState, simulation_state_buffer
+            )
```
Comment (nimazareian) [inline src/software/thunderscope/thunderscope_main.py:370]: This seems to be a duplicate of the code write above it

**Review by itsarune — COMMENTED** [review]
looks like after fixing the merge conflicts, you also have some dependency errors when you run thunderscope_main, so I'd recommend reverting the merge and re-doing it

**Comment by itsarune** [comment]
will re-open later, stale PR

**Pull request closed without merging**